Code/Framework/AzToolsFramework/Tests/Prefab/Instance/InstanceDeserializationTests.cpp

```diff
@@ -0,0 +1,205 @@
+                {
+                    // Add a transform component to entity1 of createdPrefab.
+                    entity->CreateComponent(AZ::EditorTransformComponentTypeId);
+                }
+                return true;
+            });
+
+        GenerateDomAndReloadInstantiatedPrefab(createdPrefab, instantiatedPrefab);
+
+        instantiatedPrefab->GetEntities(
+            [](const AZStd::unique_ptr<AZ::Entity>& entity)
+            {
+                if (entity->GetName() == "Entity2")
+                {
+                    // Since we didn't touch entity2 in createdPrefab, it should remain untouched in instantiatedPrefab and thus retain its
+                    // active state.
+                    EXPECT_EQ(entity->GetState(), AZ::Entity::State::Active);
+                }
+                else
+                {
+                    // Validate that the entity is in 'constructed' state, which indicates that it got reloaded.
+                    EXPECT_EQ(entity->GetState(), AZ::Entity::State::Constructed);
+                    AZStd::vector<AZ::Component*> entity1Components = entity->GetComponents();
+                    EXPECT_TRUE(entity1Components.size() == 1);
+                    AzToolsFramework::Components::TransformComponent* transformComponentInInstantiatedPrefab =
+                        reinterpret_cast<AzToolsFramework::Components::TransformComponent*>(entity1Components.front());
+
+                    // Validate that a transform component exists in entity1 of instantiatedPrefab.
+                    EXPECT_TRUE(transformComponentInInstantiatedPrefab != nullptr);
+                }
+                return true;
+            });
+    }
+
+    TEST_F(InstanceDeserializationTest, ReloadInstanceUponComponentDelete)
+    {
+        AZ::Entity* entity1 = CreateEntity("Entity1", false);
+        AzToolsFramework::Components::TransformComponent* transformComponent = aznew AzToolsFramework::Components::TransformComponent;
+        transformComponent->SetWorldTranslation(AZ::Vector3(10.0, 0, 0));
+        entity1->AddComponent(transformComponent);
+
+        AZ::Entity* entity2 = CreateEntity("Entity2", false);
+
+        AZStd::pair<InstanceUniquePointer, InstanceUniquePointer> prefabInstances =
+            SetupPrefabInstances(AzToolsFramework::EntityList{ entity1, entity2 }, m_prefabSystemComponent);
+        InstanceUniquePointer createdPrefab = AZStd::move(prefabInstances.first);
+        InstanceUniquePointer instantiatedPrefab = AZStd::move(prefabInstances.second);
+
+        createdPrefab->GetEntities(
+            [](const AZStd::unique_ptr<AZ::Entity>& entity)
+            {
+                if (entity->GetName() == "Entity1")
+                {
+                    // Remove the transform component from entity1 of createdPrefab;
+                    AZ::Component* transformComponent = entity->GetComponents().front();
+                    EXPECT_TRUE(transformComponent != nullptr);
```

Comment:
repeat: This should be EXPECT_NE
```suggestion
                    EXPECT_TRUE(nullptr, transformComponent);
```